internal/builtin/providers/tf/data_source_state.go

```diff
@@ -222,7 +223,9 @@ func getBackend(cfg cty.Value) (backend.Backend, cty.Value, tfdiags.Diagnostics)
 		config = cty.ObjectVal(config.AsValueMap())
 	}
 
-	schema := b.ConfigSchema()
+	ctx := context.TODO()
```

Comment:
Note: I'm adding these TODOs as markers where I think the host function should be accepting a context itself.